spec/unit/type/rpmkey_spec.rb

```diff
@@ -71,5 +71,16 @@
       end
     end
 
+    describe "autorequire" do
+      it "should autorequire a local file" do
+        catalog = Puppet::Resource::Catalog.new
+        file = Puppet::Type.type(:file).new(:name => '/tmp/foo', :content => 'bar' )
+        catalog.add_resource file
+        key = described_class.new(:name => 'DB42A60E', :source => '/tmp/foo', :ensure => :present)
+        catalog.add_resource key
+        key.autorequire.size.should == 1
+      end
+    end
```

Comment:
We should add to more cases:
1) source beeing a weblink and no local path to make sure we do not autorequire anything in this case
2) source beeing unset, like `described_class.new(:name => 'DB42A60E', :ensure => :absent)`